microovn/src/charm.py

```diff
@@ -68,63 +67,67 @@ def __init__(self, framework: ops.Framework):
+        err, _ = call_microovn_command("config", "set", "ovn.central-ips", ",".join(addresses))
         if err:
             logger.error("calling config set failed with code {0}".format(err))
             return False
         return True
 
-
     def _dataplane_mode(self):
-        if not self._stored.in_cluster or \
-           not self.model.get_relation(OVSDBCMD_RELATION) or \
-           not self.ovsdbcms_requires.remote_ready():
-            logger.debug("not going into dataplane mode, one of these is false in_cluster: {0}, relation_exists: {1}, remote_ready: {2}".format(self._stored.in_cluster,self.model.get_relation(OVSDBCMD_RELATION),self.ovsdbcms_requires.remote_ready()))
+        if (
+            not self._stored.in_cluster
+            or not self.model.get_relation(OVSDBCMD_RELATION)
+            or not self.ovsdbcms_requires.remote_ready()
+        ):
+            logger.debug(
+                "not going into dataplane mode, one of these is false in_cluster: {0}, relation_exists: {1}, remote_ready: {2}".format(
+                    self._stored.in_cluster,
+                    self.model.get_relation(OVSDBCMD_RELATION),
+                    self.ovsdbcms_requires.remote_ready(),
+                )
+            )
             return
 
         self.unit.status = ops.MaintenanceStatus("switching to dataplane mode")
-        err, output = call_microovn_command("disable", "central",
-                              "--allow-disable-last-central")
+        err, output = call_microovn_command("disable", "central", "--allow-disable-last-central")
         if err:
             if "this service is not enabled" in output:
                 logger.debug("central service already disabled")
             else:
-                logger.error("disabling central failed with error code: {0} and output: {1}".format(err,output))
+                logger.error(
+                    "disabling central failed with error code: {0} and output: {1}".format(
+                        err, output
+                    )
+                )
         if self.unit.is_leader():
             if not self._set_central_ips_config():
-                self.unit.status = ops.MaintenanceStatus(
-                    "failed switching to dataplane mode")
+                self.unit.status = ops.MaintenanceStatus("failed switching to dataplane mode")
                 return False
         self.unit.status = ops.ActiveStatus()
 
     def _update_status(self, _: ops.EventBase):
         if self._stored.broken_dataplane:
-            self.unit.status = ops.BlockedStatus("relation with external ovn central broken, cluster in unstable and blocked state")
+            self.unit.status = ops.BlockedStatus(
+                "relation with external ovn central broken, cluster in unstable and blocked state"
+            )
 
     def _on_ovsdbcms_broken(self, _: ops.EventBase):
-        self._stored.broken_dataplane=True
+        self._stored.broken_dataplane = True
```

Comment:
How would the human operator get out of this situation?

Imagine they deploy microovn with an external relation, regrets it and wants to use MicroOVN snap central services instead.

They would remove the relation, get this message and then log onto one of the units to amend the situation. Let's say they enable central services on three nodes using the MicroOVN snap CLI.

Then everything would be peachy right, except for the charm still thinking something is wrong.]

Is there a better way? Does the MicroOVN CLI have a light weight non-error prone way of determining whether central services are running anywhere, and could it be used to determine this state?